### PR TITLE
[Typing] Initial Pcap typing

### PR DESCRIPTION
**Pull request by ankenyr**

### CODE REVIEW

File: aerleon/lib/pcap.py [hunk 1]
```diff
@@ -192,7 +198,7 @@ def __str__(self):
 
         return cond + '\n'
 
-    def _CheckAddressAf(self, addrs):
+    def _CheckAddressAf(self, addrs: List[Union[Any, IPv6, IPv4]]) -> List[Union[str, IPv6, IPv4]]:
```
Comment (jtwb) [inline aerleon/lib/pcap.py:201]: That 'Any' can be removed, I think.

File: aerleon/lib/pcap.py [hunk 2]
```diff
@@ -206,7 +212,7 @@ def _CheckAddressAf(self, addrs):
         return af_addrs
 
     @staticmethod
-    def JoinConditionals(condition_list, operator):
+    def JoinConditionals(condition_list: List[Union[Any, str]], operator: str) -> str:
```
Comment (jtwb) [inline aerleon/lib/pcap.py:215]: Same, this Any can be removed

File: aerleon/lib/pcap.py [hunk 3]
```diff
@@ -228,7 +234,9 @@ def JoinConditionals(condition_list, operator):
         res = '(%s)' % (op.join(condition_list))
         return res
 
-    def _GenerateAddrStatement(self, addrs, exclude_addrs):
+    def _GenerateAddrStatement(
+        self, addrs: List[Union[str, IPv6, IPv4]], exclude_addrs: List[Any]
```
Comment (jtwb) [inline aerleon/lib/pcap.py:238]: `exclude_addrs` should have the same type as `addrs`, right?

File: aerleon/lib/pcap.py [hunk 4]
```diff
@@ -266,7 +276,7 @@ def _GeneratePortStatement(self, ports, direction):
                 conditions.append('%s portrange %s-%s' % (direction, port_tuple[0], port_tuple[1]))
         return Term.JoinConditionals(conditions, 'or')
 
-    def _GenerateTcpOptions(self, options):
+    def _GenerateTcpOptions(self, options: List[Union[Any, str]]) -> str:
```
Comment (jtwb) [inline aerleon/lib/pcap.py:279]: This is just `List[str]` since it only is ever called with Term.options.

File: aerleon/lib/pcap.py (hunk 1 above)
Comment (ankenyr) [reply aerleon/lib/pcap.py:201]: Done

File: aerleon/lib/pcap.py (hunk 3 above)
Comment (ankenyr) [reply aerleon/lib/pcap.py:238]: Yup, think these were missed during the late night coding.

File: aerleon/lib/pcap.py (hunk 4 above)
Comment (ankenyr) [reply aerleon/lib/pcap.py:279]: Done

**Review by jtwb — APPROVED** [review]

**Pull request merged**